src/Sarif.Converters/PREFastConverter.cs

```diff
@@ -0,0 +1,193 @@
+        {
+            if (string.IsNullOrWhiteSpace(defect.Rank))
+                return;
+
+            result.SetProperty("rank", defect.Rank);
+        }
+
+        private void SetProbability(Defect defect, Result result)
+        {
+            if (string.IsNullOrWhiteSpace(defect.Probability))
+                return;
+
+            result.SetProperty("probability", defect.Probability);
+        }
+
+        private void GenerateCodeFlows(Defect defect, Result result)
+        {
+            if (defect.Path?.SFAs.Any() == false)
+                return;
+
+            int step = 0;
+            var locations = new List<AnnotatedCodeLocation>();
+            bool pathUsesKeyEvents = defect.Path.SFAs.Any(x => !string.IsNullOrWhiteSpace(x?.KeyEvent?.Id));
+
+            foreach (var sfa in defect.Path.SFAs)
+            {
+                var reg = new Region()
+                {
+                    StartColumn = sfa.Column + 1,
+                    StartLine = sfa.Line
+                };
+
+                Uri uri = new Uri($"{sfa.FilePath}{sfa.FileName}", UriKind.Relative);
+                PhysicalLocation fileLocation = new PhysicalLocation(uri: uri, uriBaseId: null, region: reg);
+                var annotatedCodeLocation = new AnnotatedCodeLocation
+                {
+                    PhysicalLocation = fileLocation,
+                    Step = ++step
+                };
+
+                if (pathUsesKeyEvents)
+                {
+                    if (string.IsNullOrWhiteSpace(sfa.KeyEvent?.Id))
+                    {
+                        annotatedCodeLocation.Importance = AnnotatedCodeLocationImportance.Unimportant;
+                    }
+                    else
+                    {
+                        annotatedCodeLocation.SetProperty("keyEventId", sfa.KeyEvent.Id);
+                        if (Enum.TryParse(sfa.KeyEvent.Kind, true, out AnnotatedCodeLocationKind kind))
+                        {
+                            annotatedCodeLocation.Kind = kind;
+                        }
+
+                        if (Enum.TryParse(sfa.KeyEvent.Importance, true, out AnnotatedCodeLocationImportance importance))
+                        {
+                            annotatedCodeLocation.Importance = importance;
+                        }
+
+                        if (!string.IsNullOrWhiteSpace(sfa.KeyEvent.Message))
+                        {
+                            annotatedCodeLocation.Message = sfa.KeyEvent.Message;
+                        }
+                    }
+                }
+
+                locations.Add(annotatedCodeLocation);
+            }
+
+            var codeFlow = new CodeFlow()
+            {
+                Locations = locations
+            };
+
+            result.CodeFlows = new List<CodeFlow>()
+            {
+                codeFlow
+            };
+        }
+
+        private void ExtractCategories(Defect defect, Result result)
```

Comment:
I think it should be called `SetCategories` in parallel with the other `Set` method names.

Reply:
fixed

---
In reply to: [146951094](https://github.com/Microsoft/sarif-sdk/pull/657#discussion_r146951094) [](ancestors = 146951094)